File: app/src/main/java/com/example/pearvideoclient/author/FixPagerAdapter.java
package com.example.pearvideoclient.author;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

import java.util.ArrayList;
import java.util.List;

/**
 * @Description: java类作用描述
 * @Author: jiazhu
 * @Date: 2019-08-13 11:29
 * @ClassName: FixPagerAdapter
 */
public class FixPagerAdapter extends FragmentStatePagerAdapter {

    private List<String> mTitles = new ArrayList<>();
    private List<Fragment> mFragments;

    public FixPagerAdapter(FragmentManager fm, List<Fragment> fragments) {
        super(fm);
        this.mFragments = fragments;
    }

    public void setTitles(List<String> titles) {
        this.mTitles = titles;
    }

    @Override
    public Fragment getItem(int i) {
        return mFragments.get(i);
    }

    @Override
    public int getCount() {
        return mFragments == null ? 0 : mFragments.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        if (mTitles != null && !mTitles.isEmpty()) {
            return mTitles.get(position);
        }
        return null;
    }
}

File: app/src/main/java/com/example/pearvideoclient/author/FixPagerAdapter.kt
package com.example.pearvideoclient.author

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import java.util.*

/**
 * @Description: java类作用描述
 * @Author: jiazhu
 * @Date: 2019-08-13 11:29
 * @ClassName: FixPagerAdapter
 */
class FixPagerAdapter(fm: FragmentManager, private val mFragments: List<Fragment>?)
    : FragmentStatePagerAdapter(fm) {

    private var mTitles: List<String>? = ArrayList()

    fun setTitles(titles: List<String>) {
        this.mTitles = titles
    }

    override fun getItem(i: Int): Fragment {
        return mFragments!![i]
    }

    override fun getCount(): Int {
        return mFragments?.size ?: 0
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return if (mTitles != null && !mTitles!!.isEmpty()) {
            mTitles!![position]
        } else null
    }
}
